lib/service/signals.go

```diff
@@ -83,8 +83,15 @@ func (process *TeleportProcess) WaitForSignals(ctx context.Context) error {
 				process.log.Infof("All services stopped, exiting.")
 				return nil
 			case syscall.SIGTERM, syscall.SIGKILL, syscall.SIGINT:
-				process.log.Infof("Got signal %q, exiting immediately.", signal)
-				process.Close()
+				timeout := getShutdownTimeout(process.log)
+				cancelCtx, cancelFunc := context.WithTimeout(ctx, timeout)
+				process.log.Infof("Got signal %q, exiting within %vs.", signal, timeout.Seconds())
+				go func() {
+					process.Shutdown(cancelCtx)
+					cancelFunc()
```

Comment:
nit for slightly better panic safety
```suggestion
					defer cancelFunc()
					process.Shutdown(cancelCtx)
```